packages/edit-site/src/components/editor/index.js

```diff
@@ -85,38 +89,25 @@ export default function EditSiteEditor( { isPostsList = false } ) {
 	const { canvas = 'view' } = params;
 	const isLoading = useIsSiteEditorLoading();
 	useAdaptEditorToCanvas( canvas );
+	const entity = useResolveEditedEntity();
```

Comment:
To verify my understanding: here the edited entity is no longer resolved just once in `useInitEditedEntityFromUrl` and stored into the `edit-site` store, but it's resolved directly in the `EditSiteEditor` component? The `context` etc is no longer a value in data store, but just a local variable returned from a hook?

Reply:
> I think this hook and the current component can be simplified even further by moving this "route specific" logic to the routes directly.

Does that mean that we would break up the hook into multiple ones, each handling a specific case? 1) post types without a parent template; 2) pages; 3) everything else? That's how the current (very convoluted) logic seems to be structured.

Reply:
> Does that mean that we would break up the hook into multiple ones, each handling a specific case? 1) post types without a parent template; 2) pages; 3) everything else? That's how the current (very convoluted) logic seems to be structured.

Yes exactly, I want to try that next, but my fear is that the "remounting" is going to prevent me from actually doing it.

> To verify my understanding: here the edited entity is no longer resolved just once in useInitEditedEntityFromUrl and stored into the edit-site store, but it's resolved directly in the EditSiteEditor component? The context etc is no longer a value in data store, but just a local variable returned from a hook?

You can see that I still call `useSyncDeprecatedEntityIntoState` after the "resolve" hook. That one is mostly there for backwards compatibility (people still accessing the state)

Reply:
> my fear is that the "remounting" is going to prevent me from actually doing it.

Yes, instead of always rendering `<Editor />` we'd have multiple components:
```js
function TemplateEditor() {
  const { postType, postId, templateId } = useResolveEditedTemplate();
  return <Editor postType postId templateId />;
}

function PageEditor() {
  const { postType, postId, templateId } = useResolveEditedPage();
  return <Editor postType postId templateId />;
}
```
and that causes remounting.

One fact that could help is that the resolution code doesn't need to be a hook, it can be an async function that calls a series of `await resolveSelect.getEntityRecords()` and returns a promise of resolved template + context. Then the editor component always calls the same hook but with a different "resolver":
```js
const { postType, postId, templateId } = useResolve( templateFetch );
return <Editor {...} />;
```

The downside of this is that the "resolver" is not reactive: if the edited entity slug changes, for example, the React component won't be notified and updated, although the resulting template depends on the slug.

Reply:
Oh this reminds of the concept of "loaders" :) Maybe we can introduce it in our routes and just have `{ postType, postId, context }` = useLoaderData or something like that :P